structlog/stdlib.py

```diff
@@ -149,6 +198,31 @@ def __call__(self, *args):
         return logging.getLogger(name)
 
 
+class StdlibFormatEventRenderer(object):
+    """
+    Applies stdlib-like string formatting to the `event` key with the arguments
+    in the `positional_args` key. This is populated by
+    `structlog.stdlib.BoundLogger` or can be manually set.
+
+    `positional_args` can be any iterable, but a dictionary as the single
+    element of the tuple is used instead of the tuple, to mantain compatibility
+    with the undocumented feature of stdlib logging.
+
+    """
+    def __init__(self, strip_positional_args=False):
+        self.strip_positional_args = strip_positional_args
+
+    def __call__(self, _, __, event_dict):
+        args = event_dict.get('positional_args')
+        if args:
+            args = tuple(args)
+            if len(args) == 1 and isinstance(args[0], dict) and args[0]:
```

Comment:
This code implements the same formatting logic as the stdlib's `logging` module does, which is basically applying the `%` operator, which supports both an iterable (tuple) and mapping (dict) argument.

The tuple conversion is needed in the current version of the code because the `%` operator is picky when it comes to lists:

```python
>>> '%d %d %s' % [1, 2, 'test']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: %d format: a number is required, not list
>>> '%d %d %s' % (1, 2, 'test')
'1 2 test'
```

But actually `tuple()` conversion above is not needed at all. `positional_args` is always a tuple since it comes from an `*args` function argument. However, the tests for this formatter provide a list instead, so with those updated the conversion can be removed.